### PR TITLE
[docker] Add safety-rules image build

### PR DESCRIPTION
**Pull request by mgorven**
## Motivation

Package the new safety-rules binary.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Built image.

### CODE REVIEW

File: docker/safety-rules/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+RUN cargo build --release -p libra-node -p cli -p config-builder -p safety-rules && cd target/release && rm -r build deps incremental
+
+### Production Image ###
+FROM debian:buster AS prod
+
+RUN addgroup --system --gid 6180 libra && adduser --system --ingroup libra --no-create-home --uid 6180 libra
+
+RUN mkdir -p /opt/libra/bin
+COPY --from=builder /libra/target/release/safety-rules /opt/libra/bin
```
Comment (mgorven) [inline docker/safety-rules/Dockerfile]: Is this the correct binary and built target?

Comment (davidiw) [reply docker/safety-rules/Dockerfile]: Excellent!

File: docker/validator/Dockerfile [hunk 2]
```diff
@@ -16,7 +16,7 @@ FROM toolchain AS builder
 
 COPY . /libra
 
-RUN cargo build --release -p libra-node -p cli -p config-builder && cd target/release && rm -r build deps incremental
+RUN cargo build --release -p libra-node -p cli -p config-builder -p safety-rules && cd target/release && rm -r build deps incremental
```
Comment (davidiw) [inline docker/validator/Dockerfile]: why add here too?

File: docker/safety-rules/Dockerfile (hunk 1 above)
Comment (davidiw) [reply docker/safety-rules/Dockerfile]: Yes!

File: docker/client/client.Dockerfile [hunk 3]
```diff
@@ -16,7 +16,7 @@ FROM toolchain AS builder
 
 COPY . /libra
 
-RUN cargo build --release -p libra-node -p cli -p config-builder && cd target/release && rm -r build deps incremental
+RUN cargo build --release -p libra-node -p cli -p config-builder -p safety-rules && cd target/release && rm -r build deps incremental
```
Comment (davidiw) [inline docker/client/client.Dockerfile]: why do we build all these things for client?

File: docker/validator/Dockerfile (hunk 2 above)
Comment (mgorven) [reply docker/validator/Dockerfile]: This is to enable Docker layer caching so when CI builds all the images the build only happens once.

**Pull request closed without merging**